core/management/commands/backfill_commits.py

```diff
@@ -0,0 +1,63 @@
+import logging
+
+from django.core.management.base import BaseCommand, CommandError, CommandParser
+from redis import Redis
+from shared.config import get_config
+
+from core.models import Commit
+from services.task import TaskService
+
+log = logging.getLogger(__name__)
+
+
+def get_storage_redis():
+    return Redis.from_url(get_config("services", "redis_url"))
+
+
+def get_celery_redis():
+    return Redis.from_url(get_config("serivces", "celery_broker"))
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser: CommandParser) -> None:
+        parser.add_argument("--batch-size", type=int)
+
+    def handle(self, *args, **options):
+        batch_size = options.get("batch_size", 1000)
+
+        storage_redis = get_storage_redis()
+        celery_redis = get_celery_redis()
+
+        queue_name = "backfill_commits"
+        queue_length = celery_redis.llen(queue_name)
+
+        if queue_length > 0:
+            log.warning(
+                "Backfill commits queue not drained",
+                extra=dict(
+                    queue_length=queue_length,
+                ),
+            )
+            return
+
+        commits = Commit.objects.all()
+
+        # this stores the oldest commit id that has already been backfilled
+        commit_id = storage_redis.get("backfill_commits_id")
+        if commit_id:
+            commit_id = int(commit_id)
+            commits = commits.filter(id__lt=commit_id)
+
+        commits = commits.order_by("-id")[:batch_size]
```

Comment:
Can the batch size be added as a limit to the query? So we only fetch the commits we want from the Database.

Also if we can not fetch the entire commit, that would be ideal.

Reply:
See https://github.com/codecov/codecov-api/pull/102#discussion_r1316161249

Reply:
> Also if we can not fetch the entire commit, that would be ideal.

Oh yeah, good idea - will just fetch the `id` and `commitid`